k4EDM4hep2LcioConv/src/k4EDM4hep2LcioConv.cpp

```diff
@@ -11,11 +11,17 @@ lcio::LCCollectionVec* convTracks(
   auto* tracks = new lcio::LCCollectionVec(lcio::LCIO::TRACK);
 
   // Loop over EDM4hep tracks converting them to lcio tracks
+  
   for (const auto& edm_tr : (*tracks_coll)) {
     if (edm_tr.isAvailable()) {
       auto* lcio_tr = new lcio::TrackImpl();
-
-      lcio_tr->setTypeBit(edm_tr.getType());
+      // The Type of the Tracks need to be set bitwise in LCIO since the setType(int) function is private for the LCIO TrackImpl and only a setTypeBit(bitnumber) function can be used to set the Tzpe bit by bit.
+      int type = edm_tr.getType();
+      for (int i = 0; i<sizeof(int)*8;i++){
+	if (type & (1<<i)){
+	   lcio_tr->setTypeBit(i);	  
```

Comment:
```suggestion
        lcio_tr->setTypeBit(i, (type & (1 << i)));
```
you can even get rid of the branch. but I am not entirely sure whether this is more readable